### PR TITLE
Provide endpoints for Operatives data

### PR DESCRIPTION
**Pull request by alex-demetriou**
## Trello
 - ####  [Add operatives tables to the db](https://trello.com/c/Rd7NUWCJ)
 - ####  [As a user I can add an operative to a work order so that they are associated with that WO](https://trello.com/c/1PPQVvM4)

## Describe this PR
 - Adds Operative endpoints
 - Modifies the Operatives table to house data from DRS

## Progress
- [x] List endpoint
- [x] Get endpoint
- [x] Delete endpoint (soft delete)
- [x] Table migrations
- [x] Register classes in `Startup.cs`
- [ ] End-to-end tests (?)

### CODE REVIEW

File: RepairsApi/V2/Boundary/Request/Operative.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+namespace RepairsApi.V2.Boundary.Request
+{
+    public class Operative
```
Comment (josh-dadswell-hackney) [inline RepairsApi/V2/Boundary/Request/Operative.cs:3]: Can we rename this class to OperativeRequest, will also save the 'using alias'

File: RepairsApi/V2/Boundary/Response/Operative.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+using System.Collections.Generic;
+
+namespace RepairsApi.V2.Boundary.Response
+{
+    public class Operative
```
Comment (josh-dadswell-hackney) [inline RepairsApi/V2/Boundary/Response/Operative.cs:5]: Probably should rename this also operativeResponse

File: RepairsApi/V2/Controllers/OperativesController.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        /// <summary>
+        /// Retrieves all matching operatives given the query params
+        /// </summary>
+        /// <param name="operativeSearchParams">The search parameters object to match against</param>
+        /// <response code="200">Operatives found</response>
+        [HttpGet]
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(List<Operative>), 200)]
+        [ProducesResponseType(typeof(string), StatusCodes.Status502BadGateway)]
```
Comment (josh-dadswell-hackney) [inline RepairsApi/V2/Controllers/OperativesController.cs:34]: I dont think this endpoint will ever return a 502 as it doesnt use any external APIs

Comment (kylealexanderhackney) [inline RepairsApi/V2/Controllers/OperativesController.cs:34]: 502s are only returned if this endpoint calls another endpoint which can fail. I dont believe this work does that

File: RepairsApi/V2/Gateways/OperativeGateway.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+        public OperativeGateway(RepairsContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IEnumerable<Operative>> GetByQueryAsync(Boundary.Request.Operative searchModel)
+        {
+            var query = _context.Operatives
+                .Where(operative => searchModel.Id == null || operative.Id == searchModel.Id)
```
Comment (kylealexanderhackney) [inline RepairsApi/V2/Gateways/OperativeGateway.cs:22]: We have a framework for providing a filter framework on queries that simplifies this class and therefore test maintenence.

Im happy to jump into a call about how it works and work order list is a good reference
ListWorkOrdersUseCase line 29 is the start of how its used
and FilteringServiceCollectionExtensions is how it is configured

Comment (alex-demetriou) [reply RepairsApi/V2/Gateways/OperativeGateway.cs]: I've refactored to use the Filtering Service - could you have a look? :pray:

**Review by kylealexanderhackney — APPROVED** [review]
All looks good. Only thing i would mention is that an integration test that archives and then gets operatives to prove the archive works would be helpful.

**Comment by alex-demetriou** [comment]
> All looks good. Only thing i would mention is that an integration test that archives and then gets operatives to prove the archive works would be helpful.

Cheers :pray: 
Would that be significantly different to [this](https://github.com/LBHackney-IT/repairs-api-dotnet/blob/21695983915cde05ddb3768438c9c475066b85b6/RepairsApi.Tests/V2/Gateways/OperativeGatewayTests.cs#L52-L70)?

File: RepairsApi/V2/Infrastructure/Operative.cs [hunk 5]
```diff
@@ -1,11 +1,14 @@
...
 using System.ComponentModel.DataAnnotations;
+using Microsoft.EntityFrameworkCore;
 
 namespace RepairsApi.V2.Infrastructure
 {
-    public class Operative
+    public class Operative : IArchivable
     {
         [Key] public int Id { get; set; }
+        public string PayrollNumber { get; set; }
```
Comment (josh-dadswell-hackney) [inline RepairsApi/V2/Infrastructure/Operative.cs:10]: Is there any reason the payroll number is not the primary key?

**Pull request merged**